### PR TITLE
fix material disappear

### LINKED ISSUE #9544: Infinite aluminum loop is required for Rhodium
### Your GTNH Discord Username

Riking#6902

### Your Pack Version

2.1.2.2

### Your Server

Private

### Type of Server

Vanilla Forge

### Your Expectation

Building the platinum processing line, working on Rhodium. Quest book instructs us to make Sodium Nitrate.

Sodium Nitrate is combined with the Rhodium Salt Solution to get Rhodium Nitrate Dust, and as this is a recipe that has to happen a lot, Sodium Nitrate is made from base materials: Nitric Acid and Sodium.

### The Reality

The quest book goal lists `gt.bwMetaGenerateddust.82.name` instead of "Sodium Nitrate".

Sodium Nitrate cannot be made from Nitric Acid and Sodium Dust - but it _can_ be made from Nitric Acid and Sodium Carbonate Dust.

Sodium Carbonate Dust can be made from the algae farm line, or the new infinite aluminum loop line (Sodium Aluminate Dust).

Infinite Aluminum loop:

 1. Chemical Reactor (x24): Water + 1 NaOH Dust + 1 Aluminum Dust => Sodium Aluminate + Hydrogen
 2. EBF: 24 Sodium Aluminate + 3 Sodium Hydroxide + 60 B CO<sub>2</sub> gas => 24 Alumina Dust + 2 Sodium Carbonate
    Sodium Carbonate: **Output** for Rhodium processing, combine with Nitric Acid.
 3. EBF (x12): 2 Alumina Dust + 1 Carbon Dust => 3 Aluminum Ingot
 4. Macerator (x24): 1 Aluminum Ingot => 1 Aluminum Dust
   12x Aluminum Ingot: **Output** for loop. Or void barrel it.

### Your Proposal

One of the following:

1. Update the Quest Book to document that the aluminum production loop is a necessary part of rhodium processing.
2. Make an easier recipe for Sodium Nitrate.
3. Make an easier recipe for Sodium Carbonate Dust. (Something consuming Calcium Carbonate?)

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [ ] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

### PR DESCRIPTION
**Pull request by GlodBlock**
fix https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/9544
make Werkstoff won't stop generate item when a item with the same oredict exist, and add them to `GT_OreDictUnificator.addAssociation` so it can accept other items with the same oredict.

### CODE REVIEW

**Comment by Dream-Master** [comment]
last crash
https://gist.github.com/Dream-Master/28d9430bb551289b89c5742a0adf2e47
@GlodBlock

**Comment by mitchej123** [comment]
@Elisis You're the resident werkstoff expert

File: src/main/java/com/github/bartimaeusnek/bartworks/common/loaders/StaticRecipeChangeLoaders.java [hunk 1]
```diff
@@ -262,6 +263,24 @@ private static void runUnficationDeleter(Werkstoff werkstoff) {
...
+            if (werkstoff.getBridgeMaterial() != null) {
+                werkstoff.getBridgeMaterial().mElement = Element.get(werkstoff.getToolTip());
+                Element.get(werkstoff.getToolTip()).mLinkedMaterials = new ArrayList<>();
+                Element.get(werkstoff.getToolTip()).mLinkedMaterials.add(werkstoff.getBridgeMaterial());
+            }
+        }
+
+        for (OrePrefixes prefixes : OrePrefixes.values())
+            if (werkstoff.hasItemType(prefixes)) {
+                GT_OreDictUnificator.set(prefixes, werkstoff.getBridgeMaterial(), werkstoff.get(prefixes), true, true);
```
Comment (Elisis) [inline src/main/java/com/github/bartimaeusnek/bartworks/common/loaders/StaticRecipeChangeLoaders.java:277]: You haven't tested if getBridgeMaterial returns null here

**Comment by Elisis** [comment]
Other than that I can't see anything glaringly wrong. Though it seems some GT++ materials refuse to unify even with this code, like GT++ Zirconium (presumably if you're loading before the other mod), although with enforceUnification it works just just showing both in main NEI, and recipes

**Comment by Dream-Master** [comment]
yes works for me as well.

**Review by Dream-Master — APPROVED** [review]

**Pull request merged**